src/en/charms-deploying-advanced.md

```diff
@@ -0,0 +1,289 @@
+Machine 1 added.
+juju deploy mycharm --to 1 --series yakkety --force
+```
+
+It may be required to use `--force-series` when upgrading charms. For example,
+in a case where an application is initially deployed using a charm that
+supports `precise` and `trusty`. If a new version of the charm is released that
+only supports `trusty` and `xenial` then it will be allowed to upgrade
+applications deployed on `precise`, but only using `--force-series`, like this:
+
+```bash
+juju upgrade-charm mycharm --force-series
+```
+
+## Deploying to specific machines
+
+To deploy to specific, pre-existing machines the `--to` option is used. When
+this is done, unless the machine was created via `add-machine`, a charm has
+already been deployed to the machine.  
+
+!!! Note:
+    When multiple charms are deployed to the same machine there exists the
+    possibility of conflicting configuration files (on the machine's
+    filesystem). Work is being done to rectify this.
+
+Machines are often referred to by their ID number. This is a simple integer
+that is shown in the output to `juju status` (or `juju machines`). For example,
+this partial output shows a machine with an ID of '2':
+
+```no-highlight
+Machine  State    DNS           Inst id        Series  AZ  Message
+2        started  10.132.70.65  juju-79b3aa-0  xenial      Running
+```
+
+### deploy --to
+
+To deploy the 'haproxy' application to machine '2' we would do this:
+
+```bash
+juju deploy --to 2 haproxy
+```
+
+Below, the `--constraints` option is used during controller creation to ensure
+that each workload machine will have enough memory to run multiple
+applications. MySQL is deployed as the first unit (in the
+'default' model) and so ends up on machine '0'. Then Rabbitmq gets deployed to
+the same machine:
+
+```bash
+juju bootstrap --constraints="mem=4G" localhost lxd
+juju deploy mysql
+juju deploy --to 0 rabbitmq-server
+```
+
+Juju treats a container like any other machine so it is possible to target
+specific containers as well. Here we deploy to containers in two different
+ways:
+
+```bash
+juju deploy nginx --to 24/lxd/3
+juju deploy mongodb --to lxd:25
+```
+
+In the first case above, nginx is deployed to existing container '3' on machine
+'24'. In the second case, MongoDB is deployed to a **new** container on machine
+'25'. The latter is an exception to the rule that the `--to` option is always
+used in conjunction with a pre-existing machine.
+
+It is also possible to deploy units using *placement directives* as arguments
+to the `--to` option. Placement directives are cloud-specific:
+
+```bash
+juju deploy mysql --to zone=us-east-1a
+juju deploy mediawiki --to host.mass
+```
+
+The first example deploys to a specific AWS zone while the second example
+deploys to a named machine in MAAS.
+
+### add-unit --to
+
+The `add-unit` command also supports the `--to` option, including placement
+directives.
+
+<!-- THE EXAMPLE DOES NOT WORK AND THE PURPOSE OF THE TEXT IS NOT CLEAR.
```

Comment:
I don't see the point of purposefully adding something we aren't happy with and then exposing it in a comment.

The purpose of add-unit  --to is for situaions e.g. to mitigate for failure, where multiple nodes are running many services. So in what way does it not work?

Reply:
```
juju deploy --constraints="mem=4G" openstack-dashboard
Located charm "cs:openstack-dashboard-258".
Deploying charm "cs:openstack-dashboard-258".

juju add-unit --to 1 rabbitmq-server
ERROR application "rabbitmq-server" not found (not found)
```

Reply:
well, I imagine that's because openstack-dashboard doesn't include rabbit-mq. It doesn't mean that add-unit --to is broken, just that this example doesn't work because the openstack charms have changed

Reply:
I didn't say that `add-unit` was broken. I said the example was broken. And because it was, I couldn't understand the surrounding text. Now I think I get it. The purpose is to show a charm (a bundle?) that installs multiple applications (e.g. openstack-dashboard and rabbitmq-server) distributed across multiple machines, and to place an additional application unit (e.g. rabbitmq-server) on one of those machines where it currently does not exist. It's fairly convoluted because the user needs to know that the original application installs multiple sub-applications and which machine does not already contain the application-to-be-added. I personally don't think it's a great example. It has more the potential to confuse than the potential to clarify. If you have another legitimate example I can include it otherwise I think we can just omit one altogether.

Reply:
no, it's a rubbish example. But we don't really need to invent some scenario with real charms where it makes sense, we just tell people why they may want to do it, then show them how. Use nginx and postgres or something